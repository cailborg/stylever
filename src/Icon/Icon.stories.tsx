import React from "react";
import { Story, Meta } from "@storybook/react/types-6-0";
import { Icon, IconProps } from "./Icon";

export default {
  title: "Components/Icon",
  component: Icon,
  argTypes: {
    name: { control: "text" },
    index: { type: "number", min: 0, max: 9, step: 1 },
  },
} as Meta;

const Template: Story<IconProps> = (args) => <Icon {...args} />;

export const Standard = Template.bind({});
Standard.args = {
  name: "Cross",
  index: 9,
};
