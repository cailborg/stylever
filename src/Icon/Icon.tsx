import React from "react";
import styled from "styled-components";
import data from "./iconsConfig.json";
import { theme, Theme, ColorIndex } from "../theme";

export const IconDisplayName = "Icon";

export interface IconProps {
  name: keyof typeof data;
  color?: keyof Theme["colors"];
  index?: ColorIndex;
  label?: string;
}

const getIconConfig = (name: keyof typeof data) => {
  if (data.hasOwnProperty(name)) {
    return data[name];
  }
};

const StyledSVG = styled.svg`
    fill: ${props => props.color};
  `;
  
export const Icon: React.FC<IconProps> = ({
  name,
  color = "Neutral",
  label,
  index,
}) => {
  const values = getIconConfig(name) as { drawn: string; title: string };
  return (
    <StyledSVG
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      role="img"
      color={theme.color(color, index)}
    >
      <title>{label ? label : values.title}</title>
      <path d={values.drawn} fillRule="evenodd" clipRule="evenodd" />
    </StyledSVG>
  );
};
