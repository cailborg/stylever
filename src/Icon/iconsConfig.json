{
  "alert": {
    "title": "alert",
    "drawn": "M13 1a1 1 0 00-2 0v2.018c-1.656.119-2.841.834-3.649 1.928-.834 1.13-1.204 2.582-1.436 3.904-.101.577-.181 1.162-.256 1.71l-.036.264a22.87 22.87 0 01-.28 1.707c-.111.508-.237.887-.382 1.155-.14.257-.265.355-.365.4-1.777.785-2.472 2.086-2.662 3.227a4.635 4.635 0 00.106 1.97l.005.014.002.005v.003c0 .001.001.002.953-.305l-.952.307A1 1 0 003 20h6c0 1.69 1.481 3 3 3 1.518 0 3-1.31 3-3H21a1 1 0 00.951-.693L21 19c.951.307.952.306.952.306v-.004l.003-.005.004-.015.005-.017a4.367 4.367 0 00.119-.577 4.635 4.635 0 00-.017-1.375c-.19-1.14-.885-2.442-2.662-3.228-.091-.04-.228-.145-.377-.48-.153-.343-.27-.822-.372-1.446-.07-.43-.125-.878-.184-1.36v-.002c-.027-.22-.055-.448-.086-.684-.095-.73-.212-1.509-.4-2.257a11.861 11.861 0 00-.284-.962 1 1 0 10-1.891.652c.087.253.165.52.235.8.159.63.264 1.314.357 2.025l.078.628c.062.502.125 1.012.201 1.48.109.668.258 1.354.518 1.94.265.593.688 1.183 1.396 1.496 1.103.487 1.408 1.185 1.498 1.726.021.125.03.246.033.359H14a1 1 0 00-1 1v1c0 .519-.519 1-1 1-.482 0-1-.481-1-1v-1a1 1 0 00-1-1H3.873c.003-.113.013-.234.033-.358.09-.542.396-1.24 1.498-1.727.621-.275 1.035-.759 1.316-1.277.275-.508.45-1.097.577-1.68.128-.589.22-1.231.308-1.861l.037-.272c.075-.55.149-1.093.243-1.63.223-1.269.533-2.327 1.075-3.061C9.45 5.471 10.18 5 11.5 5H12A1 1 0 0013 4V1z"
  },
  "icon2": {
    "title": "icon2",
    "drawn": "M13 1a1 1 0 10-2 0v2.019c-.706.053-1.332.231-1.83.592-.654.476-.937 1.156-.995 1.845-.055.651.084 1.34.28 1.982.2.653.485 1.336.784 1.996.2.446.404.874.598 1.284.41.864.782 1.649 1.02 2.348.175.513.222.837.208 1.013a.578.578 0 01-.003.021c-1.453.643-3.57 1.64-5.322 2.474a572.485 572.485 0 00-2.928 1.406l-.184.089-.048.023-.013.006-.003.002L3 19l-.437-.9A1 1 0 003 20h6c0 1.69 1.482 3 3 3 1.519 0 3-1.31 3-3H21a1 1 0 00.951-.693L21 19c.951.307.952.306.952.306v-.004l.003-.005.004-.015.006-.02.006-.02c.01-.033.02-.076.034-.13a4.636 4.636 0 00.062-1.8c-.19-1.14-.886-2.441-2.663-3.227-.091-.04-.228-.145-.377-.48-.153-.343-.27-.822-.371-1.446-.07-.43-.125-.878-.185-1.36-.027-.221-.055-.45-.086-.686-.095-.73-.212-1.509-.4-2.257a11.861 11.861 0 00-.284-.962 1 1 0 00-1.891.652c.087.253.165.52.235.8.159.63.264 1.314.357 2.025l.078.628c.062.502.125 1.012.201 1.48.109.668.258 1.354.518 1.94.265.593.689 1.183 1.397 1.496 1.102.487 1.407 1.185 1.497 1.726.021.125.03.246.033.359H14a1 1 0 00-1 1v1c0 .519-.518 1-1 1-.481 0-1-.481-1-1v-1a1 1 0 00-1-1H7.4c1.573-.746 3.289-1.548 4.504-2.085.342-.152.633-.386.84-.705.2-.312.287-.65.314-.965.05-.6-.11-1.245-.307-1.823-.277-.811-.732-1.773-1.168-2.694-.182-.384-.36-.761-.522-1.117-.286-.635-.53-1.227-.693-1.759-.167-.543-.224-.948-.2-1.228.02-.243.092-.332.177-.394.128-.093.447-.23 1.155-.23H12A1 1 0 0013 4V1z"
  }
}