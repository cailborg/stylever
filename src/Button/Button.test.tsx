import React from "react";
import { Icon } from "../Icon/Icon";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Icon name="alert" />).toJSON();
  expect(tree).toMatchSnapshot();
});
