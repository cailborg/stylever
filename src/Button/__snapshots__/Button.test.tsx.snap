// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-AxirZ jloIDr"
  onBlur={[Function]}
  onFocus={[Function]}
  style={
    Object {
      "boxShadow": "none",
    }
  }
>
  Some button text
</button>
`;
